Reneiski_Homework1/ReneiskiSelenium11/Tests/Tests.cs

```diff
@@ -0,0 +1,106 @@
+            Assert.AreEqual("39", webTablePage.GetCellText(ConstantsWebTablePage.ColumnNameAge));
+            Assert.AreEqual("cierra@example.com", webTablePage.GetCellText(ConstantsWebTablePage.ColumnNameEmail));
+            Assert.AreEqual("10000", webTablePage.GetCellText(ConstantsWebTablePage.ColumnNameSalary));
+            Assert.AreEqual("Insurance", webTablePage.GetCellText(ConstantsWebTablePage.ColumnNameDepartament));
+
+            GenericPages.WebTablePage.ClickAddButton();
+
+            GenericPages.RegFormPage.EnterFirstName("Alex");
+            GenericPages.RegFormPage.EnterLastName("Ren");
+            GenericPages.RegFormPage.EnterUserEmail("Ren@tut.by");
+            GenericPages.RegFormPage.EnterAge("41");
+            GenericPages.RegFormPage.EnterSalary("20000");
+            GenericPages.RegFormPage.EnterDepartment("QA");
+
+            GenericPages.WebTablePage.ClickSubmitButton();
+            Assert.AreEqual(FilledTableRawTextResult, webTablePage.ReturnTextResult());
+
+            GenericPages.WebTablePage.ClickEditButton();
+            GenericPages.RegFormPage.ClearSalary();
+            GenericPages.RegFormPage.EnterEditedSalary("22000");
+            GenericPages.WebTablePage.ClickSubmitButton();
+            Assert.AreEqual(EditedTableRawTextResult, webTablePage.ReturnTextResult());
+
+            GenericPages.WebTablePage.ClickDeleteButton();
+        }
+
+        [Test]
+        public void ForthButtonsTest()
+        {
+            var buttonPage = new ButtonPage();
+            GenericPages.ButtonPage.GoToButtonsPage();
+            GenericPages.ButtonPage.DoubleClick();
+            Assert.AreEqual("You have done a double click", buttonPage.ReturnDoubleClickTextResult());
+
+            GenericPages.ButtonPage.RightClick();
+            Assert.AreEqual("You have done a right click", buttonPage.ReturnRightClickTextResult());
+
+            GenericPages.ButtonPage.DynamicClick();
+            Assert.AreEqual("You have done a dynamic click", buttonPage.ReturnDynamicClickTextResult());
+        }
+
+        [Test]
+        public void FifthLinksTest()
+        {
+            var linksPage = new LinksPage();
+            GenericPages.LinksPage.GoToLinksPage();
+            GenericPages.LinksPage.ClickHomeLink();
+            WebDriverExtensions.SwitchToTheLastWindow();
+            Assert.AreEqual(TestSettings.PageUrl, linksPage.GetUrlAttribute());
+
+            WebDriverExtensions.SwitchToTheFirstWindow();
+            GenericPages.LinksPage.ClickCreatedLink();
+            Assert.AreEqual("Link has responded with staus 201 and status text Created", linksPage.ReturnCreatedLinkTextResult());
+
+            GenericPages.LinksPage.ClickNoContentLink();
+            var isNoContentButtonResultText = WebDriverFactory.Driver.GetWebDriverWait().Until(drv => linksPage.ReturnCreatedLinkTextResult() == ConstantsLinksPage.ClickedNoContentLinkExpectedMessage);
```

Comment:
on the test class level write a separate method returning bool variable for WebDriverFactory.Driver.GetWebDriverWait().Until(drv => linksPage.ReturnCreatedLinkTextResult() == ConstantsLinksPage.ClickedNoContentLinkExpectedMessage); and call this method in the test

Reply:
Fixed